nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/folder-es-reindex/folder-es-reindex.component.ts

```diff
@@ -174,53 +177,36 @@ export class FolderESReindexComponent implements OnInit, OnDestroy {
   onReindexFormSubmit(): void {
     if (this.folderReindexForm?.valid) {
       this.elasticSearchReindexService.spinnerStatus.next(true);
-      const userInput = this.folderReindexForm
-        ?.get("documentID")
-        ?.value?.trim();
-      this.buildDocumentCountFetchRequestQuery(userInput);
+      this.userInput = this.removeLeadingCharacters(
+        this.folderReindexForm?.get("documentID")?.value.trim()
+      );
+      /* The single quote is decoded and replaced with encoded backslash and single quotes, to form the request query correctly
+          for elasticsearch reindex endpoint, for paths containing single quote e.g. /default-domain/ws1/Harry's-file will be built like
+          /default-domain/workspaces/ws1/Harry%5C%27s-file
+          Other special characters are encoded by default by nuxeo js client, but not single quote */
+      this.decodedUserInput = decodeURIComponent(this.userInput).replaceAll(
+        "'",
+        "%5C%27"
+      );
+      const requestQuery = `${ELASTIC_SEARCH_LABELS.SELECT_BASE_QUERY} ecm:uuid='${this.decodedUserInput}' OR ecm:ancestorId='${this.decodedUserInput}'`;
+      this.fetchNoOfDocuments(requestQuery);
     }
   }
 
-  buildDocumentCountFetchRequestQuery(userInput: string | null): void {
-    this.nuxeo
-      .repository()
-      .fetch(userInput)
-      .then((document: unknown) => {
-        if (
-          typeof document === "object" &&
-          document !== null &&
-          "uid" in document
-        ) {
-          const doc = document as { uid: string };
-          const documentId = doc?.uid ? doc?.uid : "";
-          if (documentId) {
-            const requestQuery = `${ELASTIC_SEARCH_LABELS.SELECT_BASE_QUERY} ecm:uuid='${documentId}' OR ecm:ancestorId='${documentId}'`;
-            this.fetchNoOfDocuments(requestQuery, documentId);
-          }
-        }
-      })
-      .catch((err: unknown) => {
-        this.elasticSearchReindexService.spinnerStatus.next(false);
-        if (this.checkIfErrorHasResponse(err)) {
-          return (
-            err as { response: { json: () => Promise<unknown> } }
-          ).response.json();
-        } else {
-          return Promise.reject(ELASTIC_SEARCH_LABELS.UNEXPECTED_ERROR);
-        }
-      })
-      .then((errorJson: unknown) => {
-        if (typeof errorJson === "object" && errorJson !== null) {
-          this.store.dispatch(
-            ReindexActions.onFolderReindexFailure({
-              error: errorJson as HttpErrorResponse,
-            })
-          );
-        }
-      });
+  removeLeadingCharacters(input: string): string {
```

Comment:
duplicate utility function used accross multiple components. Move it to a utility file in the common areas